### PR TITLE
cmd/atlas/internal/cmdapi: support multi to urls for migrate diff

### PR DESCRIPTION
**Pull request by rotemtam**

### CODE REVIEW

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 1]
```diff
@@ -699,7 +703,7 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
 		}
 		return migrate.Realm(realm), nil
 	default: // database connection
-		client, err := sqlclient.Open(ctx, MigrateFlags.ToURL)
+		client, err := sqlclient.Open(ctx, MigrateFlags.ToURLs[0])
 		if err != nil {
```
[the remark this answers is not in the document]
Comment (masseelch) [reply cmd/atlas/internal/cmdapi/migrate.go:707]: nvm, didn't see the case in `selectScheme`

**Review by masseelch — APPROVED** [review]

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 2]
```diff
@@ -716,6 +720,29 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
+	var previous string
+	if len(urls) == 0 {
+		return "", errors.New("at least one to url is required")
+	}
+	for _, url := range urls {
+		parts := strings.SplitN(url, "://", 2)
+		switch current := parts[0]; {
+		case previous == "":
+			previous = current
+			continue
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:736]:
```suggestion
```

previous/current is misleading here. You have one scheme and it's the first one (it's not updated).

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 3]
```diff
@@ -716,6 +720,29 @@ func to(ctx context.Context, client *sqlclient.Client) (migrate.StateReader, err
...
+		case previous != current:
+			return "", fmt.Errorf("got mixed url schemes: %q and %q", previous, current)
+		case current != "file":
+			return "", fmt.Errorf("got multiple urls of scheme: %q", current)
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:740]: These errors are presented to the user on failure. Use friendly ones.

File: cmd/atlas/internal/cmdapi/migrate.go [hunk 4]
```diff
@@ -232,7 +232,7 @@ func init() {
 	cobra.CheckErr(MigrateApplyCmd.MarkFlagRequired(migrateFlagURL))
 	// Diff flags.
 	urlFlag(&MigrateFlags.DevURL, migrateFlagDevURL, "", MigrateDiffCmd.Flags())
-	urlFlag(&MigrateFlags.ToURL, migrateFlagTo, "", MigrateDiffCmd.Flags())
+	MigrateDiffCmd.Flags().StringSliceVarP(&MigrateFlags.ToURLs, migrateFlagTo, "", nil, "to URLs")
```
Comment (a8m) [inline cmd/atlas/internal/cmdapi/migrate.go:235]: This changes the help output to be less friendly:
image

Let's do something like this instead?

```
[driver://username:password@address/dbname?param=value ...] select a desired state using the URL format
```

**Review by a8m — APPROVED** [review]

**Pull request merged**